### ISSUE TITLE
cPanel - How to disable CTRL-F override in phpMyAdmin?

### ISSUE THREAD

**Issue by that-ben** [phpmyadmin/phpmyadmin#17291]
A recent phpMyAdmin update (or perhaps a browser a update?) enabled the CTRL-F shortcut to just ditch whatever SQL query I'm typing  in the SQL tab and bring the SEARCH tab on instead.  This is the most retarded functionality I've seen in a long while.  How the heck can I disable this "functionality" so that phpMyAdmin never, ever switch tabs on me and lose the work I'm doing ever again? And how does a software designer come up with such a genius idea in the first place? It just blows my mind how stupid this is.

**Comment by ibennetch** [comment]
Hi,

Which operating system, web browser, web browser version, and phpMyAdmin versions are you using, please?

I've tested this and don't think this behavior is coming from phpMyAdmin, but I'd like to test more to make sure, so if you can provide more information about your installation that will help my investigation.

**Comment by ibennetch** [comment]
Also, if I've made any chances to the SQL tab, trying to navigate to another page prompts me first to confirm that I want to discard changes; did you see that prompt before the page changed?

**Comment by that-ben** [comment]
Hi, thanks for replying.  There is no warning or dialog window of any kind.  It just switches to the SEARCH tab and loses everything I'm typing in the other tab with no option to have it back.  Very frustrating.

Issue experienced under W7, Firefox 95.0.2, phpMyAdmin 4.9.7 that comes with cPanel.

**Comment by williamdes** [comment]
For the record I could not find anything related to this in the diff between the versions, maybe cPanel did some changes
https://github.com/code-lts/phpmyadmin-releases/compare/upstream/version/english/4.9.2..upstream/version/english/4.9.7

**Comment by that-ben** [comment]
It's very well possible, given the downhill quality of 2021 cPanel updates.  They broke other unrelated functionality a few months ago, so I would not be surprised, tough but their change log does not mention phpMyAdmin, except only once where they supposedly fixed a bug preventing access to it from a trial cPanel license, which may or may not have broken parts of phpMyAdmin 4.9.7... who knows?

To be fair, in Firefox, under PMA 4.9.7 (and even 4.9.2) there are A LOT of Javascript warnings in the console.  Things like these, which may be related to the issue I'm talking about:
1) JQMIGRATE: jQuery.fn.bind() is deprecated jquery-migrate.js:99:12
2) JQMIGRATE: jQuery.isWindow() is deprecated jquery-migrate.js:99:12
3) Empty string passed to getElementById(). LoginHelper.jsm:1460:8

The funny thing is it does not ALWAYS do it.  Like right now, if I hit CTRL-F, nothing unusual happens.  Sometimes, it just brings the SEARCH tab on, constantly.  Right now, it doesn't and everything behaves normally.

**Comment by williamdes** [comment]
Hi @that-ben 
I am closing this one since I can not find anything that confirms that we would override <kbd>CTRL+F</kbd>
Can you try a more recent phpMyAdmin ?

**Comment by that-ben** [comment]
It's been more than 1 year so... At some point there was an update that fixed it but I can't recall the version number exactly, but it was a very long time ago, like probably not long after posting this question.  Works fine now.  Thx.

**Comment by williamdes** [comment]
> It's been more than 1 year so... At some point there was an update that fixed it but I can't recall the version number exactly, but it was a very long time ago, like probably not long after posting this question. Works fine now. Thx.

Okay, upstream c-panel then ?

**Comment by that-ben** [comment]
Oh yes, both the update that caused the issue and the one that fixed it were coming from cPanel.

**Comment by that-ben** [comment]
OMG it's back.  Hitting CTRL-F toggles phpMyAdmin's SEARCH tab.  This is so annoying! Can this be disabled? It drives me nuts!

**Comment by nxmndr** [comment]
Same problem, first time I see this today. I'm using basic PMA 5.2.1 (no cpanel or similar) in a well-known VM (Laravel Homestead).

**Comment by svinc33** [comment]
same problem, just appeared

**Comment by that-ben** [comment]
Yep, me too.  Almost 1 year later, I still see this problem from time to time, but much more rarely nowadays, maybe once per 2 months.  I don't know what the F is causing this, but it's the most annoying thing when it happens.

**Comment by M393** [comment]
Looks like the f key is handled here:
https://github.com/phpmyadmin/phpmyadmin/blob/f2de7c9a1632cb83bff1f5a1b4a59d11f6d948ab/resources/js/src/shortcuts_handler.ts#L95-L108

The required combination to go to the search tab is either `t` -> `f` 
or `d` ->`f` when no table is select.

**Comment by williamdes** [comment]
> @williamdes Why did you close this stating that nothing in PMA is triggering this tab to be clicked when CTRL-F is pressed, then? I don't understand how you can just close this GitHub issue like that based on wrong assumptions.

I am the issue manager, so I close all the issues what should be closed.
that being said, there is some recent messages here that seems to indicate it can be reproduced
Also, I also share the idea that disabling all key bindings should be a new setting https://github.com/phpmyadmin/phpmyadmin/issues/13798
I never used them myself, and can relate wanting to disable them.
Just like the drag and drop code can be a pain when moving stuff around.

Finally, I did close the issue because the reported agreed to it https://github.com/phpmyadmin/phpmyadmin/issues/17291#issuecomment-1463986413
So, it would be better to open a new issue with a good and reproducible method of triggering this nasty bug so it can be fixed.

I am pretty sure I did have this issue on my mobile phone, maybe in the search bar.

**Comment by that-ben** [comment]
The reason for this is because although many users reported this, nobody knows exactly how to reproduce it, but I'm sure if you keep your PMA tab open for a day and use it every now and then, you will see it happening too.  What I wrote in #17291 is not valid in the end as it is still happening even with a more recent version.

**Comment by that-ben** [comment]
Thanks @williamdes for doing the right thing ❤️

**Comment by williamdes** [comment]
> The question remains: Is there a way to completely DISABLE this SEARCH tab via CTRL-F "functionality" ? Like is there a line somewhere in the config files that I could change to make PMA stop switching tabs on key combos? Overriding OS or basic browser keyboard shortcuts that have uniformly served a single purpose for more than 2 decades is a huge no-no, IMO.

As I was commenting on https://github.com/phpmyadmin/phpmyadmin/issues/13798#issuecomment-2597192417
You can disable all shortcuts: https://docs.phpmyadmin.net/en/latest/config.html#cfg_DisableShortcutKeys if that helps

**Comment by williamdes** [comment]
I have a patch that could help fix this, is there anyone that has a method to reproduce the CTRL+f bug most of the time ?
